psa-crypto/src/operations/asym_signature.rs

```diff
@@ -75,10 +75,9 @@ pub fn sign_hash(
             &mut signature_length,
         )
     })
-    .to_result()?;
-
+    .to_result();
     key.close_handle(handle)?;
-
+    sign_res?;
```

Comment:
For these cases where you *need* to do a cleanup operation that may fail (after some other core operation might've failed in the first place) we have to decide: if both fail, which error do we want to be returned? I'd say the error on the "core" operation should take precedence, in which case the two branches (as in, if the core operation failed or not) will be different

Reply:
Also, I wish we had a better approach to this kind of problem where you have some cleanup to do whenever you exit the function. In this case it's fine since there's not much branching, but in some cases it might end up being a pain having to call that `close_handle` for each possible path before doing `?`.

Reply:
I originally had it returning the first operation error, then the close handle error but after a brief discussion with @hug-dev, it was changed to how it is now. I think the reason was the [PSA API states keys aren't used](https://armmbed.github.io/mbed-crypto/html/appendix/history.html#changes-between-1-0-beta-3-and-1-0-0?highlight=psa_key_handle_t), IDs are, so it should be getting removed at some point (soon? 😄) anyway.

Reply:
Yes, I think that makes sense for this case, my comment was more general - we should have a statement somewhere as to how to prioritize these errors throughout the library. I'd even say that this applies to Parsec too. The last error to occur is not necessarily the one we should return.